proposal/more-math-functions.md

```diff
@@ -0,0 +1,445 @@
+    * [`round()`](#round)
+  * [Comparison](#comparison)
+    * [`clamp()`](#clamp)
+    * [`max()`](#max)
+    * [`min()`](#min)
+    * [`percentage()`](#percentage)
+  * [Division](#division)
+    * [`div()`](#div)
+    * [`mod()`](#mod)
+  * [Exponentiation](#exponentiation)
+    * [`log()`](#log)
+    * [`pow()`](#pow)
+    * [`sqrt()`](#sqrt)
+  * [Trigonometry](#trigonometry)
+    * [Procedures](#procedures)
+      * [Verify Inputs](#verify-inputs)
+    * [`cos()`](#cos)
+    * [`sin()`](#sin)
+    * [`tan()`](#tan)
+    * [`acos()`](#acos)
+    * [`asin()`](#asin)
+    * [`atan()`](#atan)
+    * [`atan2()`](#atan2)
+      * [Edge cases](#edge-cases)
+  * [Units](#units)
+    * [`compatible()`](#compatible)
+    * [`is-unitless()`](#is-unitless)
+    * [`unit()`](#unit)
+  * [Util](#util)
+    * [`abs()`](#abs)
+    * [`hypot()`](#hypot)
+    * [`random()`](#random)
+
+## Variables
+
+### `e`
+
+```
+$e
+```
+
+### `pi`
+
+```
+$pi
+```
+
+## Functions
+
+### Approximation
+
+#### `ceil()`
+
+```
+ceil($number)
+```
+This function is also available as a global function named `ceil()`.
+
+#### `floor()`
+
+```
+floor($number)
+```
+
+This function is also available as a global function named `floor()`.
+
+#### `round()`
+
+```
+round($number)
+```
+
+This function is also available as a global function named `round()`.
+
+### Comparison
+
+#### `clamp()`
+
+```
+clamp($min, $number, $max)
+```
+
+* Return `$min` if `$min` >= `$max`
+* Return `$min` if `$number` <= `$min`.
+* Return `$max` if `$number` => `$max`.
+* Return `$number`.
+
+#### `max()`
+
+```
+max($numbers...)
+```
+
+This function is also available as a global function named `max()`.
+
+#### `min()`
+
+```
+min($numbers...)
+```
+
+This function is also available as a global function named `min()`.
+
+#### `percentage()`
+
+```
+percentage($number)
+```
+
+This function is also available as a global function named `percentage()`.
+
+### Division
+
+#### `div()`
+
+```
+div($dividend, $divisor)
+```
+
+This function is also available as a global function named `divide()`.
+
+* Throw error if the units of `$dividend` and `$divisor` are not
+  [compatible](#compatible).
+* Return `$dividend` / `$divisor`.
+
+#### `mod()`
+
+```
+mod($dividend, $modulus)
+```
+
+* Throw error if the units of `$dividend` and `$modulus` are not
+  [compatible](#compatible).
+* Return `$dividend` % `$modulus`.
+
+### Exponentiation
+
+#### `log()`
+
+```
+log($number)
+```
+* Throw error if `$number` has units or `$number` < 0.
+* Return -Infinity if `$number` is 0.
+* Return Infinity if `$number` is infinite.
+* Return the natural log of `$number`.
+
+```
+log($number, $base)
+```
+
+* Throw error if `$base` < 0 or `$base` is 0 or `$base` is 1.
+* Return [log](#log)(`$number`) / [log](#log)(`$base`).
+
+#### `pow()`
+
+```
+pow($base, $exponent)
+```
+
+* Throw error if `$base` or `$exponent` has units.
+
+* If `$exponent` is 0:
+  * Return 1.
+
+* Else if `$exponent` is infinite:
```

Comment:
I'd explicitly compare `$exponent` to `Infinity` and `-Infinity` here, just to avoid the term "infinite". A reader will *probably* know what it's referring to, but they could wrongfully assume that it's a more complex term of art.

Reply:
Fixed here and throughout